lumicks/pylake/confocal.py

```diff
@@ -0,0 +1,241 @@
+        **kwargs
+            Forwarded to `~matplotlib.pyplot.plot`.
+        """
+        for color in ["red", "green", "blue"]:
+            self._plot_color(color, **kwargs)
+
+    @property
+    def has_fluorescence(self) -> bool:
+        return self.json["fluorescence"]
+
+    @property
+    def has_force(self) -> bool:
+        return self.json["force"]
+
+
+class ConfocalImage():
+
+    def _has_incorrect_start(self, timeline_start, timeline_dt):
+        """Checks whether the scan or kymograph starts before the timeline information. If this is the case, it will
+        lead to an incorrect reconstruction. If implemented, resolve the problem."""
+
+        # Workaround for a bug in the STED delay mechanism which could result in scan start times ending up within
+        # the sample time.
+        if timeline_start - timeline_dt < self.start < timeline_start:
+            self.start = timeline_start
+
+        if timeline_start > self.start:
+            self._fix_incorrect_start()
+            return True
+
+    def _fix_incorrect_start(self):
+        raise NotImplementedError
+
+    def _ordered_axes(self):
+        """Returns axis indices in spatial order"""
+        return sorted(self.json["scan volume"]["scan axes"], key=lambda x: x["axis"])
+
+    def _plot_color(self, color, **kwargs):
+        from matplotlib.colors import LinearSegmentedColormap
+
+        linear_colormaps = {
+            "red": LinearSegmentedColormap.from_list("red", colors=[(0, 0, 0), (1, 0, 0)]),
+            "green": LinearSegmentedColormap.from_list("green", colors=[(0, 0, 0), (0, 1, 0)]),
+            "blue": LinearSegmentedColormap.from_list("blue", colors=[(0, 0, 0), (0, 0, 1)]),
+        }
+
+        image = getattr(self, f"{color}_image")
+        return self._plot(image, **{"cmap": linear_colormaps[color], **kwargs})
+
+    def _plot(self, image, **kwargs):
+        raise NotImplementedError
+
+    def plot_rgb(self, **kwargs):
+        """Plot a full rbg kymograph image
+
+        Parameters
+        ----------
+        **kwargs
+            Forwarded to :func:`matplotlib.pyplot.imshow`.
+        """
+        image = self.rgb_image
+        image = image / np.max(image)
+        return self._plot(image, **kwargs)
+
+    def save_tiff(self, filename, dtype=np.float32, clip=False):
+        """Save the RGB photon counts to a TIFF image
+
+        Parameters
+        ----------
+        filename : str
+            The name of the TIFF file where the image will be saved.
+        dtype : np.dtype
+            The data type of a single color channel in the resulting image.
+        clip : bool
+            If enabled, the photon count data will be clipped to fit into the desired `dtype`.
+            This option is disabled by default: an error will be raise if the data does not fit.
+        """
+        if self.rgb_image.size > 0:
+            save_tiff(self.rgb_image, filename, dtype, clip, ImageMetadata.from_dataset(self.json))
```

Comment:
`save_tiff` and `ImageMetadata` aren't imported, so this won't work

Reply:
good catch